apps/pollbook/barcode-scanner-daemon/src/unitech_ts100_daemon.rs

```diff
@@ -0,0 +1,270 @@
+
+    // Spawn thread to accept clients
+    {
+        let clients = clients.clone();
+        thread::spawn(move || {
+            for stream in listener.incoming() {
+                match stream {
+                    Ok(sock) => {
+                        clients.lock().unwrap().push(sock);
+                    }
+                    Err(e) => {
+                        log!(event_id: EventId::SocketConnection, message: format!("UDS client manager encountered error with client: {}", e), event_type: EventType::SystemStatus, disposition: Disposition::Failure)
+                    }
+                }
+            }
+        });
+    }
+
+    let port = match init_port(PORT_NAME, BAUD_RATE) {
+        Ok(p) => p,
+        Err(e) => {
+            log!(
+                event_id: EventId::DeviceAttached,
+                message: format!("Failed to connect to USB barcode scanner: {e}"),
+                event_type: EventType::SystemStatus,
+                disposition: Disposition::Failure
+            );
+
+            // Exit and allow systemctl config to restart daemon
+            log!(
+                EventId::ProcessTerminated;
+                EventType::SystemAction
+            );
+            exit(1);
+        }
+    };
+
+    log!(
+        event_id: EventId::DeviceAttached,
+        message: format!("Listening for barcode scan data on {}...", PORT_NAME),
+        event_type: EventType::SystemStatus,
+        disposition: Disposition::Success
+    );
+
+    let mut reader = BufReader::new(port);
+    let running = Arc::new(AtomicBool::new(true));
+
+    if let Err(e) = ctrlc::set_handler({
+        let running = running.clone();
+        move || {
+            running.store(false, Ordering::SeqCst);
+        }
+    }) {
+        log!(
+            event_id: EventId::ErrorSettingSigintHandler,
+            message: e.to_string(),
+            event_type: EventType::SystemStatus,
+            disposition: Disposition::Failure
+        );
+    }
+
+    let mut buf = Vec::new();
+    let mut start = Instant::now();
+
+    // Poll for barcode scan data
+    loop {
+        if !running.load(Ordering::SeqCst) {
+            break;
+        }
+
+        if start.elapsed() > HEARTBEAT_LOG_INTERVAL {
+            start = Instant::now();
+            log!(
+                EventId::Heartbeat;
+                EventType::SystemStatus
+            );
+        }
+
+        buf.clear();
+
+        match reader.read_until(b'\r', &mut buf) {
+            Ok(_) => {
+                // Scanner is configured by default to terminate data with '\r'
+                if buf.ends_with(&[b'\r']) {
+                    buf.pop();
+                }
+
+                if buf.len() == 0 {
+                    continue;
+                }
+
+                // We expect the input sequence to start with the compliance indicator
+                // so we just ignore it and continue to read the next line
```

Comment:
This breaks encapsulation a bit. The "compliance" indicator is a string `@\n\rs` that indicates compliance with the AAMVA format. It's expected on every scan. We could

* move this down to `AamvaDocument`
* allow the code to attempt to parse the `@\n\rs` string as an AAMVA document and return an error from `::try_from`
* handle that specific error as a no-op

But that is weird in its own way. It's not really an error because we expect the string.

Reply:
As written this makes the compliance indicator optional. Is that something we want to allow?